appveyor/Convert-FileUrisToGitHubUris.ps1

```diff
@@ -0,0 +1,120 @@
+{
+  #files
+  for($j = 0; $j -lt $sarifLog.runs[$i].files.Count; $j++)
+  {
+    $sarifLog.runs[$i].files[$j].uri = Rebase-Uri($sarifLog.runs[$i].files[$j].uri)
+    #UNTESTED
+    $props = JSON.runs[$i].files[$j].PSObject.Properties
+    if($props.Name)
+    {
+      $URIReplacement = $props.Name
+      $content = $props.Value
+      $sarifLog.runs.files.PSObject.Properties.Remove($props.Name)
+      $sarifLog.runs.files | add-member -Name Rebase-Uri($URIReplacement) -Value $content -MemberType NoteProperty
+    }
+  }
+
+  #results
+  for($j = 0; $j -lt $sarifLog.runs[$i].results.Count; $j++)
+  {
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].locations.Count; $k++)
+    {
+      $location = $sarifLog.runs[$i].results[$j].locations[$k]
+      if(!$location.resultFile)
+      {
+        $sarifLog.runs[$i].results[$j].locations[$k].resultFile.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].locations[$k].resultFile.uri)
+      }
+
+      if(!$location.analysisTarget)
+      {
+        $sarifLog.runs[$i].results[$j].locations[$k].analysisTarget[$l].uri = Rebase-Uri($sarifLog.runs[$i].results[$j].locations[$k].analysisTarget[$l].uri)
+      }
+
+    }
+
+    #Codeflow
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].codeFlows.Count; $k++)
+    {
+      for($l = 0; $l -lt $sarifLog.runs[$i].results[$j].codeFlows[$k].locations.Count; $l++)
+      {
+        $sarifLog.runs[$i].results[$j].codeFlows[$k].locations[$l].physicalLocation.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].codeFlows[$k].locations[$l].physicalLocation.uri)
+      }
+    }
+
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].relatedLocations.Count; $k++)
+    {
+      $sarifLog.runs[$i].results[$j].relatedLocations[$k].physicalLocation.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].relatedLocations[$k].physicalLocation.uri)
+    }
+
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].stacks.Count; $k++)
+    {
+      $sarifLog.runs[$i].results[$j].stacks[$k].frames.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].stacks[$k].frames.uri)
+    }
+
+    for($k = 0; $k -lt $sarifLog.runs[$i].results[$j].fixes.Count; $k++)
+    {
+      $sarifLog.runs[$i].results[$j].fixes[$k].fileChanges.uri = Rebase-Uri($sarifLog.runs[$i].results[$j].fixes[$k].fileChanges.uri)
+    }
+  }
+
+  for($j=0; $j -lt $sarifLog.runs[$i].configurationNotifications.Count; $j++){
+    $sarifLog.runs[$i].configurationNotifications[$j].physicalLocation.uri = Rebase-Uri($sarifLog.runs[$i].configurationNotifications[$j].physicalLocation.uri);
+  }
+
+  for($j=0; $j -lt $sarifLog.runs[$i].toolNotifications.Count; $j++){
+    $sarifLog.runs[$i].toolNotifications[$j].physicalLocation.uri = Rebase-Uri($sarifLog.runs[$i].toolNotifications[$j].physicalLocation.uri);
+  }
+}
+
+$newExtension = ".github.sarif"
+$writeLocation = [System.IO.Path]::GetFileNameWithoutExtension($location)+$newExtension
+$sarifLog | ConvertTo-Json -depth 999 | Out-File $writeLocation
```

Comment:
[`ConvertTo-Json`](https://technet.microsoft.com/en-us/library/hh849922.aspx?f=255&MSPPError=-2147217396) does allow you to pass the `-InputObject` parameter through the pipeline, as you do here. As a matter of style -- and this is just my preference, I haven't seen this suggested anywhere -- I think that for the first command in a pipeline, the code is clearer if you pass the object on the command line (making the command name the first word on the line). So:

```
ConvertTo-Json -Depth 999 $sarifLog | Out-File $writeLocation
```